actions/admin/upgrades/upgrade_friends_private_acls.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * Convert comment annotations to entities
+ * 
+ * Run for 2 seconds per request as set by $batch_run_time_in_secs. This includes
+ * the engine loading time.
+ */
+// from engine/start.php
+global $START_MICROTIME;
+$batch_run_time_in_secs = 2;
+
+// if upgrade has run correctly, mark it done
+if (get_input('upgrade_completed')) {
+	// set the upgrade as completed
+	$factory = new ElggUpgrade();
+	$upgrade = $factory->getUpgradeFromPath('admin/upgrades/friendsprivateacls');
+	if ($upgrade instanceof ElggUpgrade) {
+		$upgrade->setCompleted();
+	}
+
+	return true;
+}
+
+// Offset is the total amount of processed users so far.
+$offset = (int) get_input('processed', 0);
+$limit = 50;
+
+$access_status = access_get_show_hidden_status();
+access_show_hidden_entities(true);
+
+$success_count = 0;
+$error_count = 0;
+
+do {
+	$dbprefix = elgg_get_config('dbprefix');
+	$options = array(
+		'type' => 'user',
+		'offset' => $offset,
+		'limit' => $limit,
+	);
+			
+	$users = elgg_get_entities($options);
+	
+	foreach ($users as $user) {
+		$f_acl = false;
+		$p_acl = false;
+		if ($user->friends_acl) {
+			$f_acl = get_access_collection($user->friends_acl); // use existing if there is one
+		}
+		if ($user->private_acl) {
+			$p_acl = get_access_collection($user->private_acl);
+		}
+		
+		if (!$f_acl) {
+			$f_acl_id = create_access_collection('acl:friends:' . $user->guid, elgg_get_site_entity()->guid);
+		} else {
+			$q = "DELETE FROM {$dbprefix}access_collection_membership
+				WHERE access_collection_id = {$f_acl->id}";
+			delete_data($q);
+			$f_acl_id = $f_acl->id;
```

Comment:
So this removes all friends from the existing friends ACL because later all friends are re-added?

If so, this part is unnecessary. ```add_user_to_access_collection()``` is able to handle users that are already members: https://github.com/Elgg/Elgg/blob/1.10/engine/classes/Elgg/Database/AccessCollections.php#L655